### PR TITLE
consistency checking + migration

### PR DESCRIPTION
**Pull request by g-harel**
1- created new db using tinydb 
2- using tinydb to record queries and time 
3- introduced new database using mongodb to initiate the migration process
4- implemented reads and writes to tinydb to introduce topten feature
5- implmented topten feature
6- implmented consistency checker
7- implemented shadow reads and writes to mongodb 
8- introduced the shallow reads and writed to the mongodb and consistency checker asasynchronous calls
9- connected the reports generated by consistency checker on an online reporting infrastructure to help us visualize the level of consistency between mongo and tiny db
10- implemened long term consistency checker
11- removed tinydb reads and writes and pushed mongo db to production

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,13 @@
 .coverage
+.idea
```
Comment (g-harel) [inline .gitignore:2]: is this file generated too?

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+.idea/
+workspace.xml
+db.json
 coverage/
 .installed.cfg
 engines.cfg
 env
+manage.sh
```
Comment (g-harel) [inline .gitignore]: is this temporary?

File: requirements.txt [hunk 3]
```diff
@@ -8,3 +8,5 @@ pyopenssl==17.4.0
 python-dateutil==2.6.1
 pyyaml==3.12
 requests[socks]==2.18.4
+tinydb
```
Comment (g-harel) [inline requirements.txt]: we should add a version here imo

File: .gitignore (hunk 1 above)
Comment (shelbaz) [reply .gitignore:2]: yea, generated by intellij ide usually. seems like a valid addition

File: .gitignore (hunk 2 above)
Comment (shelbaz) [reply .gitignore]: shouldnt be there , i just removed it.

File: requirements.txt (hunk 3 above)
Comment (shelbaz) [reply requirements.txt]: done

**Review by shelbaz — APPROVED** [review]
Looks good

**Comment by veken1199** [comment]
the tests were failing because of pep8 code formatting test

**Comment by shelbaz** [comment]
can re implement on another branch if that's the case. Needed to merge cuz it was due

**Comment by g-harel** [comment]
@veken1199 I had fixed all the pep8 issues, most recent builds were passing :+1:

**Pull request merged**